ttyplot.c

```diff
@@ -479,20 +568,25 @@ int main(int argc, char *argv[]) {
             if (tty >= select_nfds)
                 select_nfds = tty + 1;
         }
-        const bool previous_parse_succeeded = (r == (two ? 2 : 1));
-        struct timespec timeout;
-        timeout.tv_sec = 0;
-        if (previous_parse_succeeded) {
-            timeout.tv_nsec = 0;  // we may have more input pressing, let's not throttle it down
-        } else {
-            timeout.tv_nsec = 500 * 1000 * 1000;  // <=500 milliseconds for a healthy clock display
-        }
+        struct timespec timeout = { .tv_sec = 0, .tv_nsec = 500e6 };  // 500 milliseconds for refreshing the clock
         const int select_ret = pselect(select_nfds, &read_fds, NULL, NULL, &timeout, &empty_sigset);
 
-        const bool signal_received = ((select_ret == -1) && (errno == EINTR));
+        gettimeofday(&now, NULL);
+
+        // Refresh the clock on timeouts.
+        if (select_ret == 0)
+            redraw_needed = true;
 
-        if (signal_received) {
-            continue;  // i.e. skip right to signal handling
+        // Handle signals.
+        if (sigint_pending) {
+            break;
+        }
+        if (sigwinch_pending) {
+            sigwinch_pending = 0;
+            endwin();
+            initscr();
```

Comment:
@edgar-bonet I played with resizing the window down to vertical below-minimum… — we cannot drop `erase` and `refresh` here but that's what the pull request is doing, and it would be regression. Please:

```diff
diff --git a/ttyplot.c b/ttyplot.c
index 0e824e6..cc3eba4 100644
--- a/ttyplot.c
+++ b/ttyplot.c
@@ -607,6 +607,8 @@ int main(int argc, char *argv[]) {
             sigwinch_pending = 0;
             endwin();
             initscr();
+            erase();
+            refresh();
             gethw();
             redraw_needed = true;
         }
```

Reply:
Good catch! Fixed, thanks.